### PR TITLE
Hu mkt 09 report item

### LINKED ISSUE #31: HU-MKT-09-report-item
# User can report an inappropriate item or user

**Domain:** Marketplace

## User Story

As a authenticated ITESO user, I want to report an inappropriate item or user so that report items that violate platform guidelines (prohibited items, misleading descriptions, offensive content) or report inappropriate user behavior. reports include a reason category and optional description. reported items are flagged for admin review..

## Descripción detallada

Users should be able to report items that violate platform guidelines (prohibited items, misleading descriptions, offensive content) or report inappropriate user behavior. Reports include a reason category and optional description. Reported items are flagged for admin review. 
This feature is for the web platform. Priority: medium. 
This story focuses on the core functionality. Advanced features, edge cases, and optimizations may be addressed in separate stories.

## Acceptance Criteria

- [ ] Feature is accessible to authorized users
- [ ] All required functionality works as expected
- [ ] User receives appropriate feedback
- [ ] Error cases are handled gracefully

### Frontend
- Component for main UI
- Form validation (if applicable)
- Loading and error states
- Success/error notifications
- Responsive design for mobile and desktop

## Testing Notes

- Test with valid data (happy path)
- Test with invalid/missing required data
- Test with edge cases (empty strings, special characters, very long inputs)
- Test error handling and error messages
- Test loading states and async behavior
- Test with authenticated and unauthenticated users

## Test Data Required

- Test user accounts (authenticated ITESO users)
- Sample items with various categories
- Items in different states (published, draft, deleted)
- Valid input data for all required fields
- Invalid input data for validation testing
- Edge case data (boundary values, special characters)

## Potential Mocks

- Database queries for unit tests

## Dependencies & Duplication Check

**Dependencies:** None identified

**Possible duplicates:** None

## Assumptions

- User is authenticated as an ITESO community member
- Backend API follows RESTful conventions
- Frontend uses Next.js and TypeScript
- Database is PostgreSQL

### LINKED ISSUE #33: HU-MKT-11-set-item-price
# User can set a price for items listed for sale

**Domain:** Marketplace

## User Story

As a authenticated ITESO user, I want to set a price for items listed for sale so that set a price in mxn. the price field should validate numeric input, enforce a minimum value, and display the formatted price on the item card and detail page..

## Descripción detallada

When a user selects 'sale' as the transaction type during item publication, they must be able to set a price in MXN. The price field should validate numeric input, enforce a minimum value, and display the formatted price on the item card and detail page. 
This feature is for the web platform. Priority: high. 
This story focuses on the core functionality. Advanced features, edge cases, and optimizations may be addressed in separate stories.

## Acceptance Criteria

- [ ] Feature is accessible to authorized users
- [ ] All required functionality works as expected
- [ ] User receives appropriate feedback
- [ ] Error cases are handled gracefully

### Backend
- GET endpoint for resource retrieval
- Query parameters for filtering/pagination
- Response formatting and serialization

### Frontend
- Component for main UI
- Form validation (if applicable)
- Loading and error states
- Success/error notifications
- Responsive design for mobile and desktop
- API integration with backend endpoints

## Testing Notes

- Test with valid data (happy path)
- Test with invalid/missing required data
- Test with edge cases (empty strings, special characters, very long inputs)
- Test error handling and error messages
- Test loading states and async behavior
- Test with authenticated and unauthenticated users

## Test Data Required

- Test user accounts (authenticated ITESO users)
- Sample items with various categories
- Items in different states (published, draft, deleted)
- Valid input data for all required fields
- Invalid input data for validation testing
- Edge case data (boundary values, special characters)

## Potential Mocks

- Database queries for unit tests

## Dependencies & Duplication Check

**Dependencies:** None identified

**Possible duplicates:** None

## Assumptions

- User is authenticated as an ITESO community member
- Backend API follows RESTful conventions
- Frontend uses Next.js and TypeScript
- Database is PostgreSQL

### PR DESCRIPTION
**Pull request by AlanElPlatano**
## 📌 Summary
Briefly describe what this Pull Request does and why it is needed.

---

## 🔍 Type of Change
Mark the relevant option(s):

- [ ] Feature (new functionality)
- [ ] Bug fix
- [ ] Agent (development tooling)
- [ ] Documentation
- [ ] Refactor / cleanup
- [ ] CI / DevOps

---

## 🧩 Scope
Which part of the project is affected?

- [ ] Backend
- [ ] Frontend
- [ ] Agents
- [ ] Docs
- [ ] CI / Infrastructure

---

## 🤖 Agent Details (if applicable)
If this PR adds or updates an agent:

- **Agent name:**
- **Purpose of the agent:**
- **Input(s):**
- **Output(s):**
- **Example included:**  
  - [ ] Yes  
  - [ ] No (explain why)

---

## 🧪 Testing
Describe how this change was tested:

- [ ] Local execution
- [ ] Manual testing
- [ ] Unit tests
- [ ] Not applicable (explain why)

---

## 📎 Related Context
Link to related issues, discussions, or decisions (if any).

---

## ✅ Checklist
Before requesting a review, confirm that:

- [ ] Code builds and runs locally
- [ ] Changes are small and focused
- [ ] Code follows agreed standards
- [ ] Documentation was updated if needed
- [ ] No unrelated changes are included
- [ ] AI-generated content was reviewed and understood

---

## 📝 Notes for Reviewers
Anything reviewers should pay special attention to?

### CODE REVIEW

File: backend/marketplace/views/product.py [hunk 1]
```diff
@@ -374,6 +375,48 @@ def delete_reaction(self, request, pk=None):
...
+                {"detail": "No puedes reportar tu propio producto."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        serializer = ReportCreateSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        try:
+            serializer.save(product=product, reporter=request.user)
+        except IntegrityError:
```
Comment (Ferreirafc1133) [inline backend/marketplace/views/product.py:410]: agregaste el modelo Report pero no hay migracion en backend/marketplace/migrations/. el unique constraint no existe en db entonces este IntegrityError nunca se va a disparar y peor aun cualquier deploy de esto revienta al arrancar. makemigrations y commit

File: frontend/src/hooks/useReportProduct.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const reportProduct = useCallback(
+    async (productId: number, payload: ReportProductPayload): Promise<boolean> => {
+      setIsLoading(true);
+      setError(null);
+
+      try {
+        await apiClient(`/marketplace/products/${productId}/report/`, {
```
Comment (Ferreirafc1133) [inline frontend/src/hooks/useReportProduct.ts:20]: el backend regresa {detail: ...} pero apiClient en lib/api.ts lee body?.error?.message. cuando el usuario ya reporto o es el dueño le va a salir 'Error 400' en vez del mensaje que devolviste. o alineas el formato en el view a {error: {message: ...}} o parseas detail en catch

File: frontend/src/components/products/ProductDetail.tsx [hunk 3]
```diff
@@ -28,6 +30,9 @@ export default function ProductDetail({ productId }: ProductDetailProps) {
   const { product, setProduct, isLoading, error } = useProductDetail(productId);
   const [isTransactionDialogOpen, setIsTransactionDialogOpen] = useState(false);
   const [transactionNotice, setTransactionNotice] = useState<string | null>(null);
+  const [isReportDialogOpen, setIsReportDialogOpen] = useState(false);
+  const [hasReported, setHasReported] = useState(false);
```
Comment (Ferreirafc1133) [inline frontend/src/components/products/ProductDetail.tsx:34]: hasReported vive en state del componente. si el user recarga la pagina vuelve a false y le aparece el boton reportar, lo pica y el backend le tira 400 ya reportaste. o regresas hasReported desde el endpoint de detalle o guardas algo persistente. no es bloqueante pero UX feo

**Comment by Ferreirafc1133** [comment]
falta migracion del modelo Report. sin eso no sirve nada. cuando este listo dale vuelta

**Comment by AlanElPlatano** [comment]
El archivo de migración no está en el repo porque el gitignore tiene ignorada toda la carpeta de migraciones del marketplace (solo se hace commit las de social) y el modelo de Report ya estpa en la branch de dev en marketplace/models/report.py así que cuando se corra makemigrations el unique constraint se debería generar solo, solo siguiendo lo que ya está en el repo jeje

**Comment by Ferreirafc1133** [comment]
arreglaste lo del detail y el has_reported, bien. pero la migracion sigue sin estar. el modelo Report no existe en la db sin ella. es lo unico que falta

**Comment by Ferreirafc1133** [comment]
tienes razon, las migraciones estan en el gitignore a proposito. mi error. los demas cambios estan bien, mergeado

**Comment by Ferreirafc1133** [comment]
perdon por el pedo de la migracion, no chequé el gitignore antes de abrir la boca. buen pr, el report quedo solido
